src/code/z_sram_NES.c

```diff
@@ -1,67 +1,1663 @@
+                        }
+                        func_80112B40(globalCtx, phi_s1_2 & 0xFF);
+                    }
+                }
+                gSaveContext.save.inventory.items[i] = ITEM_BOTTLE;
+            }
+        }
+    }
+
+    REMOVE_QUEST_ITEM(25);
+
+    if (gSaveContext.save.playerData.health < 0x30) {
+        gSaveContext.save.playerData.health = 0x30;
+    }
+
+    if (GET_CUR_EQUIP_VALUE(0) < 3) {
+        gSaveContext.save.equips.equipment =
+            (1 << gEquipShifts[0]) | (gSaveContext.save.equips.equipment & gEquipNegMasks[0]);
+
+        if (CUR_FORM == 0) {
+            if ((((gSaveContext.save.stolenItems & 0xFF000000) >> 0x18) >= ITEM_SWORD_GILDED) ||
+                (((gSaveContext.save.stolenItems & 0xFF0000) >> 0x10) >= ITEM_SWORD_GILDED)) {
+
+                gSaveContext.save.equips.buttonItems[CUR_FORM][0] = ITEM_SWORD_GILDED;
+                gSaveContext.save.equips.equipment =
+                    (3 << gEquipShifts[0]) | (gSaveContext.save.equips.equipment & gEquipNegMasks[0]);
+            } else {
+                gSaveContext.save.equips.buttonItems[CUR_FORM][0] = 0x4D;
+            }
+        } else {
+            if ((((gSaveContext.save.stolenItems & 0xFF000000) >> 0x18) >= ITEM_SWORD_GILDED) ||
+                (((gSaveContext.save.stolenItems & 0xFF0000) >> 0x10) >= ITEM_SWORD_GILDED)) {
+                gSaveContext.save.equips.buttonItems[0][0] = ITEM_SWORD_GILDED;
+                gSaveContext.save.equips.equipment =
+                    (3 << gEquipShifts[0]) | (gSaveContext.save.equips.equipment & gEquipNegMasks[0]);
+            } else {
+                gSaveContext.save.equips.buttonItems[0][0] = ITEM_SWORD_KOKIRI;
+            }
+        }
+    }
+
+    if ((((gSaveContext.save.stolenItems & 0xFF000000) >> 0x18) == ITEM_SWORD_GREAT_FAIRY) ||
+        (((gSaveContext.save.stolenItems & 0xFF0000) >> 0x10) == ITEM_SWORD_GREAT_FAIRY)) {
+        gSaveContext.save.inventory.items[gItemSlots[0x10]] = ITEM_SWORD_GREAT_FAIRY;
+    }
+
+    if (((gSaveContext.save.stolenItems & 0xFF000000) >> 0x18) == ITEM_BOTTLE) {
+        for (i = 0; i < 6; i++) {
+            if (gSaveContext.save.inventory.items[gItemSlots[ITEM_BOTTLE] + i] == ITEM_NONE) {
+                gSaveContext.save.inventory.items[gItemSlots[ITEM_BOTTLE] + i] = ITEM_BOTTLE;
+                break;
+            }
+        }
+    }
+
+    if (((gSaveContext.save.stolenItems & 0xFF0000) >> 0x10) == ITEM_BOTTLE) {
+        for (i = 0; i < 6; i++) {
+            if (gSaveContext.save.inventory.items[gItemSlots[ITEM_BOTTLE] + i] == ITEM_NONE) {
+                gSaveContext.save.inventory.items[gItemSlots[ITEM_BOTTLE] + i] = ITEM_BOTTLE;
+                break;
+            }
+        }
+    }
+
+    gSaveContext.save.stolenItems &= 0xFFFFFF;
+    gSaveContext.save.stolenItems &= 0xFF00FFFF;
+
+    func_801149A0(5, 5);
+    func_801149A0(0xB, 0xB);
+    func_801149A0(0x11, 0x11);
```

Comment:
```suggestion
    func_801149A0(ITEM_OCARINA_FAIRY, SLOT_TRADE_DEED);
    func_801149A0(ITEM_SLINGSHOT, SLOT_TRADE_KEY_MAMA);
    func_801149A0(ITEM_LONGSHOT, SLOT_TRADE_COUPLE);
```
Uhhh, nice code here

Reply:
I have no idea what's going on here